src/main/java/org/sagebionetworks/bridge/spring/controllers/AssessmentResourceController.java

```diff
@@ -65,77 +73,81 @@ final void setAssessmentResourceService(AssessmentResourceService service) {
         }
-        
-        return service.getResources(appId, assessmentId, offsetByInt, pageSizeInt, categories, minRevisionInt,
+
+        return service.getResources(appId, ownerId, assessmentId, offsetByInt, pageSizeInt, categories, minRevisionInt,
                 maxRevisionInt, incDeletedBool);
     }
-    
+
     @PostMapping("/v1/assessments/identifier:{assessmentId}/resources")
     @ResponseStatus(HttpStatus.CREATED)
     public AssessmentResource createAssessmentResource(@PathVariable String assessmentId) {
         UserSession session = getAuthenticatedSession(DEVELOPER, STUDY_DESIGNER);
         String appId = session.getAppId();
+        String ownerId = getOwnerId(session);
 
         if (SHARED_APP_ID.equals(appId)) {
             throw new UnauthorizedException(SHARED_ASSESSMENTS_ERROR);
         }
 
         AssessmentResource resource = parseJson(AssessmentResource.class);
-        return service.createResource(appId, assessmentId, resource);
+        return service.createResource(appId, ownerId, assessmentId, resource);
     }
 
     @GetMapping("/v1/assessments/identifier:{assessmentId}/resources/{guid}")
     public AssessmentResource getAssessmentResource(@PathVariable String assessmentId, @PathVariable String guid) {
         UserSession session = getAuthenticatedSession(DEVELOPER, STUDY_DESIGNER);
         String appId = session.getAppId();
+        String ownerId = getOwnerId(session);
         
         if (SHARED_APP_ID.equals(appId)) {
             throw new UnauthorizedException(SHARED_ASSESSMENTS_ERROR);
         }
 
-        return service.getResource(appId, assessmentId, guid);
+        return service.getResource(appId, ownerId, assessmentId, guid);
     }
 
     @PostMapping("/v1/assessments/identifier:{assessmentId}/resources/{guid}")
     public AssessmentResource updateAssessmentResource(@PathVariable String assessmentId, @PathVariable String guid) {
         UserSession session = getAuthenticatedSession(DEVELOPER, STUDY_DESIGNER);
         String appId = session.getAppId();
+        String ownerId = getOwnerId(session);
         
         if (SHARED_APP_ID.equals(appId)) {
             throw new UnauthorizedException(SHARED_ASSESSMENTS_ERROR);
         }
 
         AssessmentResource resource = parseJson(AssessmentResource.class);
         resource.setGuid(guid);
-        
-        return service.updateResource(appId, assessmentId, resource);
+
+        return service.updateResource(appId, ownerId, assessmentId, resource);
     }
 
     @DeleteMapping("/v1/assessments/identifier:{assessmentId}/resources/{guid}")
     public StatusMessage deleteAssessmentResource(@PathVariable String assessmentId, @PathVariable String guid,
             @RequestParam(required = false) String physical) {
         UserSession session = getAuthenticatedSession(DEVELOPER, STUDY_DESIGNER, ADMIN);
         String appId = session.getAppId();
+        String ownerId = session.getParticipant().getOrgMembership();
```

Comment:
This can use the getOwnerId() method, and then you don't have to do the check below

Reply:
Changed